shaders/glsl/includes/reflections.glsl

```diff
@@ -2,37 +2,41 @@
 #include "../uniformShaderConstants"
 #include "random.glsl"
 
-vec4 buildRawSkyReflection(vec3 relativePosition, vec3 resultLighting)
+vec4 buildRawSkyReflection(vec3 reflectedVector, vec3 resultLighting)
 {
-	float horizonLine =  1.0 - (abs(relativePosition.y) / length(relativePosition.xyz));
-	vec3 horizonColor = resultLighting;
+	vec3 upPosition = normalize(vec3(0.0, abs(relativePosition.y), 0.0));
+
+	highp float isZenithSky = dot(reflectedVector, upPosition);
+	highp float horizonLine = max(0.0, 1.0-exp(-0.25 / abs(isZenithSky)));
+		isZenithSky = max(0.0, isZenithSky);
+
+	vec3 horizonColor = pow(resultLighting * vec3(0.5, 0.7, 1.0), vec3(0.5));
 	vec3 zenithColor = vec3(0.25, 0.5, 1.0) * length(resultLighting);
-	return vec4(mix(zenithColor, horizonColor, horizonLine), horizonLine);
+	return vec4(mix(zenithColor, horizonColor, horizonLine), isZenithSky);
 }
 
-vec3 buildSkyPlaneReflection(vec3 relativePosition, vec3 normalColor, vec4 skyLightReflected, float isRain, float roughness){
+vec3 buildSkyPlaneReflection(vec3 reflectedVector, vec4 skyLightReflected, float isRain, float roughness){
+
+	highp vec3 cldCoord = reflectedVector;
+		cldCoord /= cldCoord.y;

```

Comment:
I think we can get rid of this cuz probably reflect() already do this job. And also it is probably causing water reflection issue